packages/types/src/Text.ts

```diff
@@ -18,14 +22,31 @@ import Compact from './codec/Compact';
 //     wraps the `Balance`, `T::AccountId`, etc. The reasoning - with a "TypeString"
 //     we can nicely strip types down like "T::AcountId" -> "AccountId"
 export default class Text extends Base<string> {
-  constructor (value: Text | string = '') {
+  constructor (value: Text | string | AnyU8a = '') {
     super(
-      value instanceof Text
-        ? value.raw
-        : value.trim()
+      Text.decode(value)
     );
   }
 
+  static decode (input: any): string {
+    if (isString(input)) {
+      return input;
+    } else if (input instanceof Text) {
+      return input.raw;
+    } else if (input instanceof Uint8Array) {
+      const [offset, length] = Compact.decode(input);
+      return u8aToUtf8(input.subarray(offset, offset + length.toNumber()));
+    } else if (Array.isArray(input)) {
```

Comment:
I would propose removing Array until needed. As it stands we don't have anything coming in anymore as `[...]`